addOns/oast/src/main/java/org/zaproxy/addon/oast/callback/CallbackParam.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2017 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.oast.callback;
+
+import java.util.List;
+import org.parosproxy.paros.common.AbstractParam;
+import org.zaproxy.zap.utils.NetworkUtils;
+
+/** @author psiinon */
+public class CallbackParam extends AbstractParam {
+
+    private static final String PROXY_BASE_KEY = "oast.callback";
+
+    private static final String LOCAL_ADDRESS_KEY = PROXY_BASE_KEY + ".localaddr";
+    private static final String REMOTE_ADDRESS_KEY = PROXY_BASE_KEY + ".remoteaddr";
+    private static final String PORT_KEY = PROXY_BASE_KEY + ".port";
+
+    private static final String SECURE_KEY = PROXY_BASE_KEY + ".secure";
+
+    private String localAddress;
+    private String remoteAddress;
+    private int port;
+    private boolean secure;
+
+    public CallbackParam() {}
+
+    @Override
+    protected void parse() {
+        localAddress = getString(LOCAL_ADDRESS_KEY, "0.0.0.0");
+        remoteAddress = getString(REMOTE_ADDRESS_KEY, getDefaultAddress());
+        port = getInt(PORT_KEY, 0);
+        secure = getBoolean(SECURE_KEY, false);
+    }
+
+    private String getDefaultAddress() {
+        List<String> addrs = NetworkUtils.getAvailableAddresses(false);
+        for (String addr : addrs) {
+            if (!addr.contains(":") && !addr.equals("localhost") && !addr.equals("127.0.0.1")) {
+                // Return the first IPv4 one that's not localhost
```

Comment:
Should this deal with the whole 127.0.0.0/8? Or am I over thinking? 

(Yes I know this file was copied)